src/main/java/org/openRealmOfStars/AI/Mission/EspionageHandling.java

```diff
@@ -0,0 +1,1128 @@
+    bonuses.addBonus(newBonus, who.getRace());
+  }
+
+  /**
+   * Execute behavior that should happen when Leader gets caught
+   * during an espionage mission.
+   * @param type EspionageMission type
+   * @param planet Planet where to do espionage
+   * @param fleet Fleet who is doing the espionage
+   * @param info Realm who is doing the espionage
+   * @param game Full game information.
+   * @param espionageSucceed If true then espionage mission succeed, but now
+   *        leader will also get caught.
+   */
+  public static void handleCaughtEspionage(final EspionageMission type,
+      final Planet planet, final Fleet fleet, final PlayerInfo info,
+      final Game game, final boolean espionageSucceed) {
+    final var starmap = game.getStarMap();
+    final var infoIndex = starmap.getPlayerList().getIndex(info);
+    final var planetPlayer = planet.getPlanetPlayerInfo();
+    final var planetEmpireName = planetPlayer.getEmpireName();
+    final var spy = fleet.getCommander();
+    final var spyCallName = spy.getCallName();
+    final var spyEmpireName = info.getEmpireName();
+    final var diplomacyWithSpy = planetPlayer.getDiplomacy()
+        .getDiplomacyList(infoIndex);
+    final boolean war = planetPlayer.getDiplomacy().isWar(infoIndex);
+    final boolean tradeWar = planetPlayer.getDiplomacy()
+        .isTradeEmbargo(infoIndex);
+
+    // Texts and text templates
+    final var txtCaughtBase = String.format(
+        "%1$s from %2$s was caught by %3$s while doing espionage mission.",
+        spyCallName, spyEmpireName, planetEmpireName);
+    final var tplCaught = txtCaughtBase + " Main goal was %1$s. ";
+    final var txtEscapedPrisonWealth = String.format(
+        "%1$s was able to escape from %2$s prison time "
+            + "by using massive amount of credits. ",
+        spyCallName, planetEmpireName);
+    final var txtPrisonBase = String.format(
+        "%1$s decided to imprison %2$s",
+        planetEmpireName, spyCallName);
+    final var tplPrison = txtPrisonBase + " for %1$s.";
+    final var txtEscapedExecutionWealth = String.format(
+        "%1$s was able to escape from %2$s execution"
+            + "by using massive amount of credits. ",
+        spyCallName, planetEmpireName);
+    final var txtExecuted = String.format(
+        "%1$s was executed by %2$s.",
+        spyCallName, planetEmpireName);
+    var txtExecReasonGeneric = "";
+    if (war) {
+      txtExecReasonGeneric = " Execution was done because of "
+          + "war times.";
+    } else if (tradeWar) {
+      txtExecReasonGeneric = " Execution was done because of "
+          + "revenge of trade war.";
+    }
+
+    if (type == EspionageMission.GAIN_TRUST) {
+      if (diplomacyWithSpy != null) {
+        diplomacyWithSpy.addBonus(DiplomacyBonusType.ESPIONAGE_BORDER_CROSS,
+            planetPlayer.getRace());
+        LeaderUtility.handleLeaderReleased(info, planet, fleet,
+            txtCaughtBase + " Since it espionage mission was gaining"
```

Comment:
There is a typo(This was already in original) " Since it espionage mission was gaining" -> " Since espionage mission was gaining"